hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtTargetResource.java

```diff
@@ -302,4 +282,24 @@ private Target findTargetWithExceptionIfNotFound(final String controllerId) {
                 .orElseThrow(() -> new EntityNotFoundException(Target.class, controllerId));
     }
 
+    @Override
+    public ResponseEntity<MgmtAction> updateAction(@PathVariable("controllerId") final String controllerId,
+            @PathVariable("actionId") final Long actionId, @RequestBody final MgmtActionRequestBodyPut actionUpdate) {
+
+        Action action = deploymentManagement.findAction(actionId)
+                .orElseThrow(() -> new EntityNotFoundException(Action.class, actionId));
+        if (!action.getTarget().getControllerId().equals(controllerId)) {
+            LOG.warn("given action ({}) is not assigned to given target ({}).", action.getId(), controllerId);
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
```

Comment:
Just a hint: There is also a builder available => `ResponseEntity.notFound().build()`. But this is totally fine as well